groupnews-web/src/main/java/com/epam/internship/controller/UserController.java

```diff
@@ -25,18 +26,27 @@
 	public List<User> getUsers() {
 		return userService.getAllUsers();
 	}
-	
-	@PostMapping(value="new")
-	public ResponseEntity<User> registerUser(@RequestParam String name, @RequestParam String email){
-		if(name.equals("") || email.equals("")){
+
+	@PostMapping(value = "new")
+	public ResponseEntity<User> registerUser(@RequestParam String name, @RequestParam String email) {
+		if (name.equals("") || email.equals("")) {
 			return new ResponseEntity<>(HttpStatus.BAD_REQUEST);
 		}
-		if(userService.emailAlreadyExists(email)){
+		if (userService.emailAlreadyExists(email)) {
 			return new ResponseEntity<>(HttpStatus.CONFLICT);
 		}
-		if(userService.isEmailValid(email)){
+		if (userService.isEmailValid(email)) {
```

Comment:
This can be included in line 32, so you won't search the database for an invalid email.